iot/google/cloud/iot_v1/gapic/device_manager_client.py

```diff
@@ -381,11 +385,13 @@ def update_device_registry(self,
             metadata = []
         metadata = list(metadata)
         try:
-            routing_header = google.api_core.gapic_v1.routing_header.to_grpc_metadata(
-                [('device_registry.name', device_registry.name)], )
-            metadata.append(routing_header)
+            routing_header = [('device_registry.name', device_registry.name)]
         except AttributeError:
             pass
```

Comment:
Here, the `AttributeError` might occur, in the case that `device_registry` is a `dict`:  in that case, don't we need to fetch its `name` key?  E.g.:

```python
    try:
        registry_name = device_registry.name
    except AttributeError:
        registry_name = device_registry['name']

    routing_header = [('device_registry.name', registry_name)
    routing_metadata = google.api_core.gapic_v1.routing_header.to_grpc_metadata(
        routing_header)
    metadata.append(routing_metadata)
```